### ISSUE TITLE
Is there a simple demo of the TFmarco?

### ISSUE THREAD

**Issue by willanxywc** [AlexKuhnle/TFMacros#1]
Hello, I m from the ShapeWorld which relies on TFmarco to build models, so would it possible for a simple demo or instruction on how to use this?

**Comment by AlexKuhnle** [comment]
Hi @willanxywc,

there are the examples in the [examples](https://github.com/AlexKuhnle/ShapeWorld/tree/master/models/agreement) directory, which gives you an idea of how to use some components. Generally, instead of using any of these components, you can also just write TensorFlow code in a similar file, i.e. one with a function `def model(model, inputs, ...)` accepting some necessary arguments (`world_shape`, `vocabulary_size`, `caption_shape` in the case of agreement datasets, see [here](https://github.com/AlexKuhnle/ShapeWorld/blob/master/train.py#L65)) and whatever other config values you want to pass. The custom version requires 1-2 additional things, which I will post here soon.

**Comment by AlexKuhnle** [comment]
Hey again @willanxywc,

I added a [custom model](https://github.com/AlexKuhnle/ShapeWorld/blob/master/models/agreement/custom.py) to the agreement examples. Essentially, it is important that losses are added via `tf.losses.add_loss(loss=...)` and that `model.register_placeholder(...)` for input and `model.register_tensor(...)` for output are called with the right key value. This at least is how it currently works...

Oh, and you also need to add a json file in the subdirectory `/hyperparams/` with the hyperparameters that the model function takes as arguments (besides the mandatory ones like `model`, `inputs`, etc...), as is given for the custom model [here](https://github.com/AlexKuhnle/ShapeWorld/blob/master/models/agreement/hyperparams/custom.params.json).

**Comment by AlexKuhnle** [comment]
Hope that helps, let me know if there are any further questions.

**Comment by willanxywc** [comment]
Thx~The custom.py really helps
